test/unit/atomic/test-atomic-ref-addsub.cpp

```diff
@@ -0,0 +1,72 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-19, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+///
+/// Source file containing tests for atomic add, subtract, inc, and dec methods
+///
+
+#include <RAJA/RAJA.hpp>
+#include "RAJA_gtest.hpp"
+
+template <typename T, typename AtomicPolicy>
+void testAtomicAddSub()
+{
+  T theval = (T)0;
+  T * memaddr = &theval;
+
+  // explicit constructor with memory address
+  RAJA::AtomicRef<T, AtomicPolicy> test1( memaddr );
+
+  // test inc ops
+  T val2 = ++test1;
+  T val3 = test1++;
+  ASSERT_EQ( test1, (T)2 );
+  ASSERT_EQ( val2, (T)1 );
+  ASSERT_EQ( val3, (T)1 );
+
+  // test dec ops
+  T val4 = --test1;
+  T val5 = test1--;
+  ASSERT_EQ( test1, (T)0 );
+  ASSERT_EQ( val2, (T)1 );
+  ASSERT_EQ( val3, (T)1 );
+
+  // test add/sub ops
+  test1 += (T)23;
```

Comment:
We should probably test the result of these as well.
The op= operators return the new value.
The fetch_op methods return the old value.

Reply:
Good call. I'll make sure to test the results in other unit tests too (some of them already have it).

Reply:
looks good